### PR TITLE
fix: filter container state without option "-a"

### PR DESCRIPTION
**Pull request by xyz-li**
`docker ps -f status=exited` will filter and output exited containers.

### CODE REVIEW

**Comment by fahedouch** [comment]
Hi @xyz-li,

thank you for your PR.

as mention in the [doc](https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-blue_square-nerdctl-ps) : 

```
-a, --all: Show all containers (default shows just running)
```

either we modify the doc to say that `-f` will enable all containers by default or you just append `-a` flag. We prefer to stick to docker behavior

**Comment by xyz-li** [comment]
According to the docker implementation, it just enables all containers when specifying the filter `status`.
https://github.com/moby/moby/blob/master/daemon/list.go#L244-L251

**Comment by fahedouch** [comment]
> According to the docker implementation, it enables all containers when specifying the filter `status`. https://github.com/moby/moby/blob/master/daemon/list.go#L244-L251

let's reflect this in the [doc](https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-blue_square-nerdctl-ps)

**Comment by apostasie** [comment]
LGTM (pending documentation change)

File: docs/command-reference.md [hunk 1]
```diff
@@ -449,7 +449,7 @@ Usage: `nerdctl ps [OPTIONS]`
 
 Flags:
 
-- :whale: `-a, --all`: Show all containers (default shows just running)
+- :whale: `-a, --all`: Show all containers (default shows just running). When specifying the option '-f' with 'status', it filters through all containers
```
Comment (fahedouch) [inline docs/command-reference.md:452]: I would rather change the `-f` flag line 463

`:whale: -f, --filter: Filter all containers based on given conditions`

**Review by fahedouch — APPROVED** [review]
Thanks

**Pull request merged**